src/stateMachineFwk/actionTable.cpp

```diff
@@ -53,7 +53,7 @@ namespace SM
 		    if (nextStatep != NULL)
 		    {
 		        cb.setNextState(nextStatep);
-		        statep->OnEntry(cb);
+		        nextStatep->OnEntry(cb);
```

Comment:
Typically what do we do on entry ?

Reply:
We fetch the current state from the Control Block and start the timer for the same.